main.py

```diff
@@ -11,11 +11,14 @@ def main():
     menu = create_menu()
 
     while True:
-        choice = ui.display_menu_get_choice(menu)
-        action = menu.get_action(choice)
-        action()
-        if choice == 'Q':
-            break
+        try:
+            choice = ui.display_menu_get_choice(menu)
+            action = menu.get_action(choice)
+            action()
+            if choice == 'Q':
+                break
+        except BookError:
```

Comment:
Sorry both, I completely missed the notification for this one. No, this is not where you would catch the BookError. Good practice to catch exceptions as close as possible to where they are created, so the try-except would be better in the function that adds the book.  What if another part of the program - like, deleting a book - and this raised a book error, and the program printed "The book you have entered is already in the database".  Would that be confusing?